src/darwin/Framework/CHIP/CHIPDevice.mm

```diff
@@ -323,9 +337,25 @@ - (void)subscribeWithQueue:(dispatch_queue_t)queue
     readParams.mKeepSubscriptions
         = (params != nil) && (params.keepPreviousSubscriptions != nil) && [params.keepPreviousSubscriptions boolValue];
 
-    auto callback = std::make_unique<SubscriptionCallback>(queue, reportHandler, subscriptionEstablishedHandler);
-    auto readClient = std::make_unique<ReadClient>(InteractionModelEngine::GetInstance(), device->GetExchangeManager(),
-        callback->GetBufferedCallback(), ReadClient::InteractionType::Subscribe);
+    std::unique_ptr<SubscriptionCallback> callback;
+    std::unique_ptr<ReadClient> readClient;
+    std::unique_ptr<AttributeCache> attributeCache;
+    if (attributeCacheContainer) {
+        __weak CHIPAttributeCacheContainer * weakPtr = attributeCacheContainer;
```

Comment:
So in terms of ownership, once no one is holding on to the `CHIPAttributeCacheContainer` the subscription will be killed off, right?  Or am I misunderstanding how this all works?

Reply:
Good point. Your understanding is correct. I applied the same logic as was used with previous `CHIPAttributeCacheContainer` API: if the cache is no longer used, the subscription stops. Now with this API, there is a separate report handler associated with it, I wonder if we should continue the subscription even if the cache is not used any more. What do you think?

Reply:
Per discussion, I'll remove this behavior and keep the subscription even if `CHIPAttributeCacheContainer` object is released.